core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/store/dot-locales-list.store.ts

```diff
@@ -16,64 +39,232 @@ export interface DotLocaleRow {
+    private readonly languageService = inject(DotLanguagesService);
+    private readonly dotMessageService = inject(DotMessageService);
+    private readonly messageService = inject(MessageService);
+    private readonly dotHttpErrorManagerService = inject(DotHttpErrorManagerService);
+    private readonly dotPushPublishDialogService = inject(DotPushPublishDialogService);
+
     // Updaters
-    readonly setLocales = this.updater((state: DotLocalesListState, languages: DotLanguage[]) => ({
+
+    readonly setResolvedData = this.updater(
+        (
+            state: DotLocalesListState,
+            routeResolved: {
+                data: DotLocalesListResolverData;
+                isEnterprise: boolean;
+                pushPublishEnvironments: DotEnvironment[];
+            }
+        ) => ({
+            ...state,
+            initialLocales: routeResolved.data.locales,
+            locales: this.processLanguages(
+                routeResolved.data.locales,
+                routeResolved.isEnterprise,
+                routeResolved.pushPublishEnvironments
+            ),
+            countries: routeResolved.data.countries,
+            languages: routeResolved.data.languages,
+            isEnterprise: routeResolved.isEnterprise,
+            pushPublishEnvironments: routeResolved.pushPublishEnvironments
+        })
+    );
+
+    readonly setLocales = this.updater((state: DotLocalesListState, locales: DotLanguage[]) => ({
         ...state,
-        locales: this.processLanguages(languages)
+        initialLocales: locales,
+        locales: this.processLanguages(locales, state.isEnterprise, state.pushPublishEnvironments)
     }));
 
     readonly vm$ = this.select(
         this.state$,
-        ({ locales }): DotLocaleListViewModel => ({
-            locales
+        ({ locales, countries, languages, initialLocales }): DotLocaleListViewModel => ({
+            locales,
+            initialLocales,
+            countries,
+            languages
         })
     );
 
+    //Effects
+    readonly makeDefaultLocale = this.effect<number>((languageId$) => {
+        return languageId$.pipe(
+            switchMap((languageId) => this.languageService.makeDefault(languageId)),
+            switchMap(() => this.languageService.get()),
+            tap((languages) => {
+                this.setLocales(languages);
+                this.messageService.add({
+                    severity: 'success',
+                    summary: this.dotMessageService.get('locale.notification.success.title'),
+                    detail: this.dotMessageService.get('locale.notification.success.message')
+                });
+            }),
+            catchError((error: HttpErrorResponse) => this.dotHttpErrorManagerService.handle(error))
+        );
+    });
+
+    readonly deleteLocale = this.effect<number>((languageId$) =>
+        languageId$.pipe(
+            switchMap((languageId) => this.languageService.delete(languageId)),
```

Comment:
What happend if the network is really slow? in the UI, do you have any `Loading` status?

Reply:
Will work loading UI in other PR, but add the  state changed in the store